lib/web_ui/lib/src/engine/image_decoder.dart

```diff
@@ -379,15 +379,16 @@ class ResizingCodec implements ui.Codec {
   @override
   Future<ui.FrameInfo> getNextFrame() async {
     final ui.FrameInfo frameInfo = await delegate.getNextFrame();
-    return AnimatedImageFrameInfo(
-      frameInfo.duration,
-      scaleImage(
+    final ui.Image scaledImage = scaleImage(
         frameInfo.image,
         targetWidth: targetWidth,
         targetHeight: targetHeight,
         allowUpscaling: allowUpscaling,
-      ),
     );
+    if (!identical(frameInfo.image, scaledImage)) {
```

Comment:
This doesn't seem right to me, as we actually dispose the original image inside the image scaling code: https://github.com/flutter/engine/pull/54096/files#diff-31d1d1318cb022b618cf4b314284d2555b4219200e8e40f67c9c1bc275ac1986L469

Are we sure this is the actual issue here? Maybe I'm just missing something but it seems like we'd be double-disposing the original image. Perhaps this is just covering up some other problem we have?

Reply:
The unit test failures seem to indicate that we are double disposing now with your change, so I suspect this needs to be looked at a bit deeper.

Reply:
Looked at this again, and I think the issue is actually that `CkResizingCodec._scaleImageUsingDomCanvas` was not matching the dispose behavior of `ResizingCodec.scaleImageIfNeeded`

This is passing CI now - PTAL